Source/WebCore/css/StyleProperties.cpp

```diff
@@ -177,35 +177,44 @@ MutableStyleProperties::MutableStyleProperties(const StyleProperties& other)
     }
 }
 
-bool StyleProperties::shorthandHasVariableReference(CSSPropertyID propertyID, String& shorthandValue) const
+std::optional<String> StyleProperties::commonShorthandChecks(CSSPropertyID propertyID) const
 {
     auto shorthand = shorthandForProperty(propertyID);
-    if (shorthand.length()) {
-        size_t numSetFromShorthand = 0;
-        // Checks for shorthand property if any of its longhand properties have set to a variable
-        // or are all pending substitution
-        for (size_t i = 0; i < shorthand.length(); i++) {
-            auto cssPropertyValue =  getPropertyCSSValue(shorthand.properties()[i]);
-            
-            auto hasBeenSetFromLonghand = is<CSSVariableReferenceValue>(cssPropertyValue);
-            auto hasBeenSetFromShorthand = is<CSSPendingSubstitutionValue>(cssPropertyValue);
-            auto hasNotBeenSetFromRequestedShorthand = hasBeenSetFromShorthand && downcast<CSSPendingSubstitutionValue>(*cssPropertyValue).shorthandPropertyId() != propertyID;
-            
-            // Request for shorthand value should return empty string if any longhand values have been
-            // set to a variable or if they were set to a variable by a different shorthand.
-            if (hasBeenSetFromLonghand || hasNotBeenSetFromRequestedShorthand)
-                return true;
-            if (hasBeenSetFromShorthand)
-                numSetFromShorthand += 1;
-        }
-        if (numSetFromShorthand) {
-            if (numSetFromShorthand != shorthand.length())
-                return true;
-            shorthandValue = downcast<CSSPendingSubstitutionValue>(* getPropertyCSSValue(shorthand.properties()[0])).shorthandValue().cssText();
-            return true;
-        }
+    if (!shorthand.length())
+        return String();
+    std::optional<bool> importance;
+    size_t numSetFromShorthand = 0;
+    for (size_t i = 0; i < shorthand.length(); i++) {
+        // FIXME: Shouldn't serialize the shorthand if some longhand is missing.
+        int propertyIndex = findPropertyIndex(shorthand.properties()[i]);
+        if (propertyIndex == -1)
+            continue;
+
+        // Don't serialize the shorthand if longhands have different importance.
+        auto property = propertyAt(propertyIndex);
+        if (!importance)
```

Comment:
In my version I wrote this:
```
bool isImportant = property.isImportant()
if (importance.value_or(isImportant) != isImportant)
    return emptyString();
Importance = isImportant;
```
You may prefer the way you wrote it, but I thought I’d say anyway.

Reply:
Switched to that.